File: flutter_zoom_videosdk-2.4.0/ios/Classes/FlutterZoomVideoSdkLiveStreamHelper.m
#import "FlutterZoomVideoSdkLiveStreamHelper.h"
#import "JSONConvert.h"

@implementation FlutterZoomVideoSdkLiveStreamHelper

- (ZoomVideoSDKLiveStreamHelper *)getLiveStreamHelper {
    ZoomVideoSDKLiveStreamHelper* liveStreamHelper = nil;
    @try {
        liveStreamHelper = [[ZoomVideoSDK shareInstance] getLiveStreamHelper];
        if (liveStreamHelper == nil) {
            NSException *e = [NSException exceptionWithName:@"NoLiveStreamoHelperFound" reason:@"No Live Stream Helper Found" userInfo:nil];
            @throw e;
        }
    } @catch (NSException *e) {
        NSLog(@"%@ - %@", e.name, e.reason);
    }
    return liveStreamHelper;
}

-(void) canStartLiveStream: (FlutterResult) result {
    result([[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @([[self getLiveStreamHelper] canStartLiveStream])]);
}

-(void) startLiveStream: (FlutterMethodCall *)call withResult:(FlutterResult) result {
    dispatch_async(dispatch_get_main_queue(), ^{
        result([[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @([[self getLiveStreamHelper] startLiveStreamWithStreamingURL:call.arguments[@"streamUrl"] StreamingKey:call.arguments[@"streamKey"] BroadcastURL:call.arguments[@"broadcastUrl"]])]);
    });
}

-(void) stopLiveStream: (FlutterResult) result {
    dispatch_async(dispatch_get_main_queue(), ^{
        result([[JSONConvert ZoomVideoSDKErrorValuesReversed] objectForKey: @([[self getLiveStreamHelper] stopLiveStream])]);
    });
}

@end

File: flutter_zoom_videosdk-2.4.0/android/src/main/java/com/flutterzoom/videosdk/FlutterZoomVideoSdkLiveStreamHelper.java
package com.flutterzoom.videosdk;

import android.app.Activity;

import androidx.annotation.NonNull;

import com.flutterzoom.videosdk.convert.FlutterZoomVideoSdkErrors;

import java.util.Map;

import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import us.zoom.sdk.ZoomVideoSDK;
import us.zoom.sdk.ZoomVideoSDKLiveStreamHelper;

public class FlutterZoomVideoSdkLiveStreamHelper {

    private Activity activity;

    FlutterZoomVideoSdkLiveStreamHelper(Activity activity) {
        this.activity = activity;
    }

    private ZoomVideoSDKLiveStreamHelper getLiveStreamHelper() {
        ZoomVideoSDKLiveStreamHelper liveStreamHelper = null;
        try {
            liveStreamHelper = ZoomVideoSDK.getInstance().getLiveStreamHelper();
            if (liveStreamHelper == null) {
                throw new Exception("No Live Stream Helper Found");
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
        return liveStreamHelper;
    }

    public void canStartLiveStream(@NonNull MethodChannel.Result result) {
        result.success(FlutterZoomVideoSdkErrors.valueOf(getLiveStreamHelper().canStartLiveStream()));
    }

    public void startLiveStream(@NonNull MethodCall call, @NonNull MethodChannel.Result result) {
        Map<String, Object> param = call.arguments();
        String streamUrl = (String) param.get("streamUrl");
        String streamKey = (String) param.get("streamKey");
        String broadcastUrl = (String) param.get("broadcastUrl");

        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                result.success(FlutterZoomVideoSdkErrors.valueOf(getLiveStreamHelper().startLiveStream(streamUrl, streamKey, broadcastUrl)));
            }
        });
    }

    public void stopLiveStream(@NonNull MethodChannel.Result result) {
        activity.runOnUiThread(new Runnable() {
            @Override
            public void run() {
                result.success(FlutterZoomVideoSdkErrors.valueOf(getLiveStreamHelper().stopLiveStream()));
            }
        });
    }

}
